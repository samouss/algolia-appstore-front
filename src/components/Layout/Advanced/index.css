.Advanced {
  display: flex;
  flex-direction: column;
  margin-top: 1vh;
}

.Advanced__Content {
  margin-bottom: 1vh;
  display: none;

  @media (--tablet) {
    display: flex;
    flex-direction: row;
  }
}

.Advanced__Content--visible {
  display: flex;
  flex-direction: column;
}

.Advanced__Link {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 25px;
  color: #b9b9b9;
  font-size: 0.8rem;

  @media (--tablet) {
    display: none;
  }
}
