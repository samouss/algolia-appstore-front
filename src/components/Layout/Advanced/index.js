import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './index.css';

class Advanced extends Component {

  constructor(props) {
    super(props);

    this.state = {
      isVisible: false,
    };

    this.toggle = this.toggle.bind(this);
  }

  toggle(event) {
    event.preventDefault();

    this.setState(state => ({
      isVisible: !state.isVisible,
    }));
  }

  render() {
    const { isVisible } = this.state;
    const { children } = this.props;
    const icon = isVisible ? <i className="fa fa-chevron-up" aria-hidden="true" />
      : <i className="fa fa-chevron-down" aria-hidden="true" />;

    return (
      <div styleName="Advanced">
        <div
          styleName={cx('Advanced__Content', {
            'Advanced__Content--visible': isVisible,
          })}
        >
          {children}
        </div>
        <a
          href=""
          styleName="Advanced__Link"
          onClick={this.toggle}
        >
          {!isVisible && 'Advanced filters'}
          {icon}
        </a>
      </div>
    );
  }

}

Advanced.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Advanced;
