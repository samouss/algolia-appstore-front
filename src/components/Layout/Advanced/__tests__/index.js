import React from 'react';
import { shallow } from 'enzyme';
import Advanced from '../index';

describe('<Advanced />', () => {
  const defaultProps = {};

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Advanced
        {...props}
      >
        Content
      </Advanced>,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to turn content to visible', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Advanced
        {...props}
      >
        Content
      </Advanced>,
    );

    component
      .find('[styleName="Advanced__Link"]')
      .simulate('click', {
        preventDefault: () => {},
      });

    expect(component).toMatchSnapshot();
  });

  it('expect to turn content to hide', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Advanced
        {...props}
      >
        Content
      </Advanced>,
    );

    component.setState({
      isVisible: true,
    });

    component
      .find('[styleName="Advanced__Link"]')
      .simulate('click', {
        preventDefault: () => {},
      });

    expect(component).toMatchSnapshot();
  });
});
