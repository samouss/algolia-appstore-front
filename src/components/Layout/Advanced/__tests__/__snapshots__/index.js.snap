// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Advanced /> expect to render 1`] = `
<div
  styleName="Advanced"
>
  <div
    styleName="Advanced__Content"
  >
    Content
  </div>
  <a
    href=""
    onClick={[Function]}
    styleName="Advanced__Link"
  >
    Advanced filters
    <i
      aria-hidden="true"
      className="fa fa-chevron-down"
    />
  </a>
</div>
`;

exports[`<Advanced /> expect to turn content to hide 1`] = `
<div
  styleName="Advanced"
>
  <div
    styleName="Advanced__Content"
  >
    Content
  </div>
  <a
    href=""
    onClick={[Function]}
    styleName="Advanced__Link"
  >
    Advanced filters
    <i
      aria-hidden="true"
      className="fa fa-chevron-down"
    />
  </a>
</div>
`;

exports[`<Advanced /> expect to turn content to visible 1`] = `
<div
  styleName="Advanced"
>
  <div
    styleName="Advanced__Content Advanced__Content--visible"
  >
    Content
  </div>
  <a
    href=""
    onClick={[Function]}
    styleName="Advanced__Link"
  >
    <i
      aria-hidden="true"
      className="fa fa-chevron-up"
    />
  </a>
</div>
`;
